### PR TITLE
fix($controllerIntrospectorProvider): fix instantiating of controllers that use the array DI notation

### PR DESCRIPTION
**Pull request by nikaspran**
Not sure about the best way to fix it, but here's one attempt.

This addresses the missing DI information when instantiating a controller that uses the array notation for DI in [this line](https://github.com/angular/router/blob/master/src/router-directive.es5.js#L389).

### CODE REVIEW

**Comment by btford** [comment]
Jeez, this was a pretty bad oversight.

Needs an associated test, otherwise looks good.

**Comment by nikaspran** [comment]
Will ammend in a few hours!

File: test/router-viewport.es5.spec.js [hunk 1]
```diff
@@ -686,27 +702,27 @@ describe('ngOutlet', function () {
   });
 
 
-  function registerComponent(name, template, config) {
+  function registerComponent(name, template, componentConstructor, routeConfig) {
```
Comment (nikaspran) [inline test/router-viewport.es5.spec.js]: I feel like this function should be removed in favour of being explicit in tests - too much logic going on here.

**Comment by nikaspran** [comment]
@btford I've included a test :)

**Comment by twhitbeck** [comment]
Yay, this one just bit me. Let's get this in. thanks @nikaspran

**Comment by btford** [comment]
:+1: Looks good!

**Pull request merged**